gprofiler/profilers/java.py

```diff
@@ -305,9 +312,10 @@ def _get_stop_cmd(self, with_output: bool) -> List[str]:
         ap_params.append(f"log={self._log_path_process}")
         return self._get_base_cmd() + [",".join(ap_params)]
 
-    def _run_async_profiler(self, cmd: List[str]) -> None:
+    def _run_async_profiler(self, cmd: List[str], timeout: int) -> None:
         try:
-            run_process(cmd)
+            # kill jattach with SIGTERM if it hangs. it will go down
+            run_process(cmd, stop_event=self._stop_event, timeout=timeout, kill_signal=signal.SIGTERM)
```

Comment:
What happens if this really gets killed by a timeout / stop event? Should we call `stop_async_profiler()`?

I see `JattachException` doesn't do that even if it detects async profiler is loaded...

Reply:
Re-starting AP is handled in `_profile_ap_process`. An event of timeout is something unknown and I don't think we should run any additional operations on the target Java process